### PR TITLE
Added Web3 4.x and eth-tester support

### PR DESCRIPTION
**Pull request by voith**
### What was wrong?
populus was no longer compatible with web3 4.x. 
It did not support `eth-tester`
`eth-utils` was outdated

### How was it fixed?
updated web3 and made compatible with new web3 syntax
added `eth-tester` support and removed `ethtestrpc`
updated `eth-utils` version
depends on https://github.com/ethereum/web3.py/pull/969 and https://github.com/ethereum/populus/pull/472

### CODE REVIEW

**Comment by pipermerriam** [comment]
@voith something to think about as you attack all of these ongoing issues with Populus:  The configuration framework is a terrible mess and should probably be removed entirely.  If you find it's getting in your way as you make these changes, consider trying to rip it out entirely.

**Comment by voith** [comment]
@pipermerriam Are you talking the upgrade config API?

**Comment by voith** [comment]
Also I'm planning to drop python3.4 support.

**Comment by pipermerriam** [comment]
I'm talking about removing the entire config API in favor of something significantly simpler and less brittle.

**Comment by voith** [comment]
@pipermerriam Thank you so much for taking the time to review this. If anything goes wrong because of this PR, I'll fix it myself.

File: populus/utils/hexadecimal.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from hexbytes import HexBytes
+
+
+def hexbytes_to_hexstr(val):
```
Comment (kkom) [inline populus/utils/hexadecimal.py]: Hey @kkom here! :) I came across a problem with this function when trying to compile a Vyper contract.

The function makes the assumption that `val` can be of either type `HexBytes` or `str`, which unfortunately isn't true.

More discussion (and a proposed alternative) available where I commented about this issue first: https://github.com/voith/populus/commit/37139e5d0dce609cc6971e603da3cc1a5554701b#r30091655

Comment (voith) [reply populus/utils/hexadecimal.py]: @kkom Thanks for reporting. This is one portion that I forgot to test manually. Let me add a

Comment (kkom) [reply populus/utils/hexadecimal.py]: My local patch is good enough, as it lets me compile code (and hopefully deploy? Haven’t tested it yet.). I’m going to use some other framework for unit and integration testing in the meantime.

I’ll wait for you to merge your commits into master, and then dig around to see what I can fix. :)

Let me know if you need any help, with PR review, etc

Comment (voith) [reply populus/utils/hexadecimal.py]: @kkom https://github.com/ethereum/populus/pull/474/commits/032eb453bce753afaae5b526db5eb18d02c28c3c + https://github.com/ethereum/populus/pull/475 Should fix the problem.

I'm just waiting for someone to review #474.

Comment (voith) [reply populus/utils/hexadecimal.py]: #474 got merged!

**Pull request merged**